client/post-editor/media-modal/index.jsx

```diff
@@ -89,7 +91,7 @@ module.exports = React.createClass( {
 	isDisabled: function() {
 		return some( this.props.mediaLibrarySelectedItems, function( item ) {
 			var mimePrefix = MediaUtils.getMimePrefix( item );
-			return item.transient && ( mimePrefix !== 'image' || ModalViews.GALLERY === this.state.activeView );
+			return item.transient && ( mimePrefix !== 'image' || ModalViews.GALLERY === this.props.view );
 		}.bind( this ) );
```

Comment:
Indeed, I'll defer to circling back in a future pull request. Sometime I'll go out of my way to refactor items like these, but in this case the pull request had already grown quite large.